### PR TITLE
Fix CUDA build failures

### PR DESCRIPTION
**Pull request by mingzhe09088**
Breaking this out of #8338 

This fixes some CUDA related build and runtime issues after BUILD_CAFFE2 and BUILD_ATEN are removed.

cc @orionr

### CODE REVIEW

File: cmake/Utils.cmake [hunk 1]
```diff
@@ -278,7 +278,7 @@ function(target_enable_style_warnings TARGET)
             -Wredundant-decls
             -Wno-shadow
             -Wsign-promo
-            -Wstrict-overflow=5
+            -Wno-strict-overflow
```
Comment (orionr) [inline cmake/Utils.cmake]: @goldsborough heads up that we're changing this. @mingzhe09088 can you give a link to the related CI failure? Thanks.

**Review by orionr — APPROVED** [review]
LGTM

File: cmake/Utils.cmake (hunk 1 above)
Comment (mingzhe09088) [reply cmake/Utils.cmake]: @orionr I couldn't find the link to the related CI failure. The change was in this commit: https://github.com/pytorch/pytorch/pull/8338/commits/28e83aa637890f18cf095f93c979991560c3daaf

Comment (orionr) [reply cmake/Utils.cmake]: Probably hard to find... In that case, @goldsborough are you okay with us temporarily changing it to get BUILD_CAFFE2 and BUILD_ATEN removal landed? We can then add it back.

**Pull request closed without merging**